CDAP-13558 Support CDC pipelines on Spark version 2.x

mlozbin-cybervisiontech opened the pull request:
JIRA - https://issues.cask.co/browse/CDAP-13558
WIKI - https://wiki.cask.co/display/CE/CDC+Solution+Enhancements

In scope of this PR:
- introduced single message format
- added schema validation
- added config validation
- changed exception handling to follow recent guidelines

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- introduced single message format
- added schema validation
- added config validation
- changed exception handling to follow recent guidelines

mlozbin-cybervisiontech commented:
@albertshau, could you please take a look.

albertshau commented on src/main/java/co/cask/cdc/plugins/common/Schemes.java:
```diff
@@ -0,0 +1,56 @@
+/**
+ * Helper class with common cdc schemes definitions.
+ */
+public class Schemes {
```
nit: Schemas

mlozbin-cybervisiontech replied:
fixed

albertshau commented on src/main/java/co/cask/cdc/plugins/sink/CDCBigTableConfig.java:
```diff
@@ -0,0 +1,88 @@
+    }
+    String serviceAccountFilePath = resolveServiceAccountFilePath();
+    if (!containsMacro(SERVICE_ACCOUNT_FILE_PATH)) {
+      if (serviceAccountFilePath != null) {
```
this can be combined with the previous line to remove one layer of nesting.

mlozbin-cybervisiontech replied:
fixed

albertshau reviewed (commented):
minor comments

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- introduced single message format
- added schema validation
- added config validation
- changed exception handling to follow recent guidelines
